recorder/src/main/cpp/stacktraces.cc

```diff
@@ -0,0 +1,23 @@
+#include "stacktraces.hh"
+
+std::uint32_t Stacktraces::fill_backtrace(NativeFrame* buff, std::uint32_t capacity) {//TODO: write me 3 tests { (capacity > stack), (stack > capacity) and (stack == capacity) }
+    std::uint64_t rbp, rpc;
+    asm("movq %%rbp, %%rax;"
+        "movq %%rax, %0;"
+        "lea (%%rip), %%rax;"
```

Comment:
So, it'll have little or no effect on register allocation, because we have enough registers for this function anyway. Basically compiler is aware of output variable clobber, so it knows which registers rbp and rpc will use. %rbp can't be used as additional general purpose register in this case anyway, because we build with no-omit-frame-pointers. %rax is a general purpose register, but it wasn't marked clobbered (FL). It can use any of the other general purpose registers for rbp and rpc.

It is free to use %rax after the computation. It doesn't run short of it before the computation anyway (except in inline scenario). It'll have to overflow one variable to memory in that case, but it'll be served out of L1 anyway, so fetch latency will not affect us.